src/lib/operators/get_table.cpp

```diff
@@ -42,24 +42,52 @@ std::shared_ptr<AbstractOperator> GetTable::_on_deep_copy(
 
 void GetTable::_on_set_parameters(const std::unordered_map<ParameterID, AllTypeVariant>& parameters) {}
 
-std::shared_ptr<const Table> GetTable::_on_execute() {
+std::shared_ptr<const Table> GetTable::_on_execute() { Fail("GetTable can't be called without transaction context."); }
```

Comment:
It's a bit confusing that this read only operator can be called only with a transaction context, or that the transaction context can be `nullptr` only if there are no discarded chunks. Perhaps some more documentation on this logic would good if there is no better way to handle the required transaction context here. Also we feel that it is a bit inconsistent that you're using `Fail` here but only a `DebugAssert` below to allow `nullptr` transaction context if there are no discarded chunks.

Reply:
I had to work my way to the class hierarchy to understand what was happening. So AbstractOperator only knows `_on_execute(tx_context)`. `AbstractReadOnlyOperator` introduces `_on_execute() -> _on_execute(nullptr)` and does little more. This operator makes `_on_execute()` illegal. See what's smells here? :)

Reply:
Well, this took me some time to understand as well. And I agree with both of you, pretty strange code. It does not really make sense what we built here! 😃 (@d-justen, please also have a look)
We'll address this.